api/auth_middleware.go

```diff
@@ -24,57 +34,135 @@ func getServiceAccountID(ctx context.Context) (string, bool) {
 }
 
 // authMiddleware authenticates either access_token or key pair
-func authMiddleware(
-	sasUC usecases.ServiceAccounts,
-) func(http.Handler) http.Handler {
+func authMiddleware(sasUC usecases.ServiceAccounts) func(http.Handler) http.Handler {
 	return func(next http.Handler) http.Handler {
-		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			authorization := r.Header.Get("authorization")
-			parts := strings.Split(authorization, " ")
-			if authorization == "" || len(parts) != 2 {
-				w.WriteHeader(http.StatusUnauthorized)
+		return http.HandlerFunc(func(responseWriter http.ResponseWriter, request *http.Request) {
```

Comment:
is better to leave w and r as params, more common in golang.

Reply:
I usually prefer more descriptive names for variables (even if they are somewhat longer), but if its a common pattern in Go to name these variables like this I'm gonna change it 👍

Reply:
for very common params I think shorter names is ok, most places call them w and r. For other cases, I also prefer descriptive names.